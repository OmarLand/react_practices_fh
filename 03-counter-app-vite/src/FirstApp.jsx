import PropTypes from 'prop-types'

// Manejo de los PROPS en los componentes
export const FirstApp = ( { title , subtitle, name } ) => {

    // console.log( props )

    return (
        <>
            {/* <h1> { title } </h1> */}
            <h1 data-testid='test-title'> { title } </h1>
            <p> { subtitle } </p>
            <p> { subtitle } </p>
            <p> { name } </p>
        </>
    )

}

// Controlamos el tipo de datos que solicitamos en el componente
FirstApp.propTypes = {
    title    : PropTypes.string.isRequired,
    subtitle : PropTypes.string.isRequired,
}

// Controlamos que exista un valor por defecto en los props que metemos al componente
FirstApp.defaultProps = {
    title    : 'No hay un titulo.',
    subtitle : 'No hay un subtitulo',
    name     : 'Omar Landaeta' 
}