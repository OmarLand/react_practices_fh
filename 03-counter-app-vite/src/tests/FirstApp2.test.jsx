import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"


describe('Pruebas en FirstApp Component', () => {
    
    const title    = 'Hola Soy Omar!'
    const subtitle = 'Soy otro subtitulo de pruebas!'

    test('Debe de hacer Match con el Snapshot', () => { 

        const { container } = render( <FirstApp title={ title }/> );
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar el mensaje "Hola, soy Omar"', () => {

        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();

    });

    test('Debe de mostrar el titulo en un h1', () => {
        render( <FirstApp title = { title } /> );
        expect( screen.getByRole('heading', { level:1 }).innerHTML ).toContain( title );
    })

    test('Debe de mostrar el subtitulo enviado mediante props:', () => { 
        render(
            <FirstApp
                title = { title }
                subtitle = { subtitle }
            />
        );

        expect( screen.getAllByText(subtitle).length ).toBe(2);
    })

 })