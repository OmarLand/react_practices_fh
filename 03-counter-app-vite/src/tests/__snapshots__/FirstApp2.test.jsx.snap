// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Component Debe de hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola Soy Omar!
     
  </h1>
  <p>
     
    No hay un subtitulo
     
  </p>
  <p>
     
    No hay un subtitulo
     
  </p>
  <p>
     
    Omar Landaeta
     
  </p>
</div>
`;
